### PR TITLE
Create i18n extension and add french pack

### PR DESCRIPTION
**Pull request by sirugh**
## Description

Uses the [previous example](https://github.com/magento-research/pwa-studio-target-experiments/pull/2/files) to construct an extension that provides some maintainable translation packs.

## Related Issue

Closes [PWA-1071](https://jira.corp.magento.com/browse/PWA-1071).

### Verification Steps

1. Delete `venia-ui/i18n/fr_FR.json` if you have it. 
2. `yarn venia add -D @magento/venia-sample-language-packs@0.0.1`
3. `yarn watch:venia` using a backend with a french locale enabled.
4. Use the store switcher to change to French store view. Translations should work.

## Checklist

* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.

### CODE REVIEW

File: packages/venia-concept/package.json
Comment (sirugh) [inline packages/venia-concept/package.json:55]: We will have to publish the extension to npm before merging this PR.

Comment (davemacaulay) [reply packages/venia-concept/package.json:55]: Why would we want to provide every install with a French language pack from an example extension? I thought this language pack was to enable testing when required?

**Review by davemacaulay — DISMISSED** [review]

**Comment by larsroettig** [comment]
From my point of view should be an optional dependency and only English should be shipped by default

File: packages/venia-concept/package.json
Comment (sirugh) [reply packages/venia-concept/package.json:55]: We do need a less manual way to provide it for the production deployments of `master` and `develop` as we have added a french store view to the backends to demonstrate the functionality. I believe @dpatil-magento has uploaded a copy of the `fr_FR.json` file to the CI system somewhere but I expect he has to manually update it.

On top of that, I had assumed we would just not include this dependency when scaffolding, but I realize that not everyone uses scaffolding (some just clone/fork) so that's probably not a valid assumption.

Sounds like we need to come up with a way optionally include this dependency, but as you are supposed to include extensions as a dependency of your project, I'm not really sure how we can do that here. Maybe in the extension we can add an environment variable to only include it when necessary?

Comment (sirugh) [reply packages/venia-concept/package.json]: Silly me, moved this to an opt-in experience. @dpatil-magento can set up the CI to install it. Developers can add it as a dependency (or link it) locally. Just need to make sure we on the core team don't accidentally commit it to venia-concept's package.json!

**Review by tjwiebell — DISMISSED** [review]
Straight forward and works as expected, nice work!

FYI - you can get yarn to use your local workspace copy if you specify a version, no need to go through symlinking. This should do it: `yarn venia add @magento/i18n-defaults@0.0.1 -D`.

**Comment by chris-brabender** [comment]
I would recommend you have a french language native review the strings here before merging ;)

**Comment by sirugh** [comment]
@chris-brabender we were mostly going to use this as a way to version control the demo language pack so we have **one** place for it instead of managing several across our CI and development. I'm not sure if we have any french speakers on our team and I'm not sure if we know anyone who'd have time to double check all of these strings :D

**Comment by revanth0212** [comment]
Personally, I don't like the name `i18n-demo`. If this package is gonna provide language packs that are traditionally not shipped through Venia, how about calling it `i18n-language-pack`?

**Review by revanth0212 — DISMISSED** [review]

File: packages/extensions/venia-sample-language-packs/i18n/fr_FR.json
Comment (benoitrenaud) [inline packages/extensions/venia-sample-language-packs/i18n/fr_FR.json:181]: **"Mot de passe"**, no caps on all words.

File: packages/extensions/venia-sample-language-packs/i18n/fr_FR.json
Comment (benoitrenaud) [inline packages/extensions/venia-sample-language-packs/i18n/fr_FR.json:132]: **"Inscription par courriel"**

**Review by benoitrenaud — CHANGES_REQUESTED** [review]
There are several other corrections I haven't commented yet. The use of the word "Livraison" vs. "Expédition" should be ruled out to be consistent throughout the document.

Be careful with capitalization on each word of a statement.

File: packages/extensions/venia-sample-language-packs/i18n/fr_FR.json
Comment (jeremycharron) [inline packages/extensions/venia-sample-language-packs/i18n/fr_FR.json:27]: Second-person singular (**Tu es**) is used in a familiar way in french, eg : when speaking to a friend of family. I don't know if this was the intend here. Using second-person plural (**Vous êtes**) is usually the standard in french when we want to be polite / be professional / speaking to a stranger.

File: packages/extensions/venia-sample-language-packs/i18n/fr_FR.json
Comment (jeremycharron) [inline packages/extensions/venia-sample-language-packs/i18n/fr_FR.json:293]: "**État**" is written with "**É**" (accent aigu)

File: packages/extensions/venia-sample-language-packs/i18n/fr_FR.json
Comment (jeremycharron) [inline packages/extensions/venia-sample-language-packs/i18n/fr_FR.json:284]: **"en rupture de stock"** does not need dash between words in french.
This seems like a direct translation of "out-of-stock".

File: packages/extensions/venia-sample-language-packs/i18n/fr_FR.json
Comment (jeremycharron) [inline packages/extensions/venia-sample-language-packs/i18n/fr_FR.json:278]: "**Chariot**" should be "**Panier**"

File: packages/extensions/venia-sample-language-packs/i18n/fr_FR.json
Comment (jeremycharron) [inline packages/extensions/venia-sample-language-packs/i18n/fr_FR.json:203]: This seems like a weird translation. There is no "Check-out" in french. I'm not sure what is the intend of this label. Used on a button to go to checkout?

File: packages/extensions/venia-sample-language-packs/i18n/fr_FR.json
Comment (jeremycharron) [inline packages/extensions/venia-sample-language-packs/i18n/fr_FR.json:243]: "**Ordre**" should be "**Commande**"

File: packages/extensions/venia-sample-language-packs/i18n/fr_FR.json
Comment (jeremycharron) [inline packages/extensions/venia-sample-language-packs/i18n/fr_FR.json:260]: "**Total**" does not take a "**e**" at this end if used in a singular way

File: packages/extensions/venia-sample-language-packs/i18n/fr_FR.json
Comment (jeremycharron) [inline packages/extensions/venia-sample-language-packs/i18n/fr_FR.json:234]: When "**Suivi**" is used as a noun, it doesn't take a "**e**" at this end.

File: packages/extensions/venia-sample-language-packs/i18n/fr_FR.json
Comment (jeremycharron) [inline packages/extensions/venia-sample-language-packs/i18n/fr_FR.json:5]: Because there are multiple informations on the Account page, we always put a "**s**" to "**Informations**" in french.
In the french translations of Magento 2, there is also a "**s**".

File: packages/extensions/venia-sample-language-packs/i18n/fr_FR.json
Comment (sirugh) [reply packages/extensions/venia-sample-language-packs/i18n/fr_FR.json:176]: The uses thus far have been as the display for when a balance is "$0", so like if shipping method has a $0 option, it will display as "FREE" or "Gratuit".

File: packages/extensions/venia-sample-language-packs/i18n/fr_FR.json
Comment (sirugh) [reply packages/extensions/venia-sample-language-packs/i18n/fr_FR.json:203]: Used for the button in the mini cart that takes the shopper to the cart/checkout page.
[Image from Gyazo](https://gyazo.com/d35f08f43d982c23e58e3a7031925c8c)

File: packages/extensions/venia-sample-language-packs/i18n/fr_FR.json
Comment (sirugh) [inline packages/extensions/venia-sample-language-packs/i18n/fr_FR.json:190]: Would this be `Mode d'expedition`? @benoitrenaud

File: packages/extensions/venia-sample-language-packs/i18n/fr_FR.json
Comment (sirugh) [inline packages/extensions/venia-sample-language-packs/i18n/fr_FR.json:189]: I'm not sure how to change "livraison" to "d'expedition" here. I assume you can't use "la d'expedition".

File: packages/extensions/venia-sample-language-packs/i18n/fr_FR.json
Comment (jeremycharron) [reply packages/extensions/venia-sample-language-packs/i18n/fr_FR.json:203]: @sirugh This is actually a good dilemma as there is no literal translation for Check-Out. I looked multiple french websites and asked colleagues and here's a couple suggestions : 
"Paiement" (my favorite for a small button, could also be used as a title for checkoutPage.titleCheckout)
"Procéder au paiement"
"Passer à la caisse"

File: packages/extensions/venia-sample-language-packs/i18n/fr_FR.json
Comment (jimbo) [inline packages/extensions/venia-sample-language-packs/i18n/fr_FR.json:29]: Given that you changed the `errorOffline` conjugation to be more polite, we probably want to make the same change throughout.

```suggestion
    "app.infoOnline": "Vous êtes en ligne.",
```

**Review by jimbo — DISMISSED** [review]

File: packages/extensions/venia-sample-language-packs/i18n/fr_FR.json
Comment (benoitrenaud) [reply packages/extensions/venia-sample-language-packs/i18n/fr_FR.json]: You can use : **"Informations sur l'expedition"** @sirugh

File: packages/extensions/venia-sample-language-packs/i18n/fr_FR.json
Comment (benoitrenaud) [reply packages/extensions/venia-sample-language-packs/i18n/fr_FR.json]: Maybe more : "Méthode d'expédition" @sirugh

**Comment by dpatil-magento** [comment]
QA Approved.

**Review by supernova-at — APPROVED** [review]

**Pull request merged**
